studio/src/mas-repository.js

```diff
@@ -91,21 +97,22 @@ export class MasRepository extends LitElement {
     update() {
         super.update();
         if (!this.foldersLoaded.value) return;
-        /**
-         * Automatically fetch data when search/filters update.
-         * Both load methods have page guards (ex. fragments won't be searched on the 'welcome' page)
-         */
-        if (this.page.value === 'content') {
-            this.searchFragments();
-        }
-        if (this.page.value === 'welcome') {
-            this.loadRecentlyUpdatedFragments();
+        this.handleSearch();
+    }
+
+    handleSearch() {
```

Comment:
I honestly don't like this approach, this operations should be handled in a single source of truth. That's why I created [router.js](https://github.com/adobecom/mas/blob/7d06c1b592458985c396581b6408457cccb2cd86/studio/src/router.js) in my PR.

Reply:
I'm waiting for your implementation. 
If your PR goes first, I'll just merge it in.